File: quicksort/quicksort.go
package quicksort

func quicksort(nums []int) []int {
	if len(nums) < 2 {
		return nums
	}

	pivot := nums[0]
	lower := make([]int, 0)
	higher := make([]int, 0)

	for i := 1; i < len(nums); i++ {
		if nums[i] <= pivot {
			lower = append(lower, nums[i])
		} else if nums[i] > pivot {
			higher = append(higher, nums[i])
		}
	}

	return append(quicksort(lower), append([]int{pivot}, quicksort(higher)...)...)
}

File: quicksort/quicksort.py
#!/usr/bin/python3

from typing import List

def quicksort(nums: List[int]) -> List[int]:
    if len(nums) < 2:
        return nums

    pivot = nums[0]

    lower = []
    higher = []

    for i in range(1, len(nums)):
        if nums[i] <= pivot:
            lower.append(nums[i])
        elif nums[i] > pivot:
            higher.append(nums[i])

        i += 1

    return quicksort(lower) + [pivot] + quicksort(higher)
